### PR TITLE
Fix chapter progress circle

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

The next chapter on the full-screen player has a progress circle around it. This was broken recently when I upgraded the chapters tab to Compose. This change fixes that and cleans up some of the chapter code.

Fixes https://github.com/Automattic/pocket-casts-android/issues/1424

## Testing Instructions

1. Play a Podcast with chapters (for example [Upgrade](https://pca.st/upgrade))
2. Advance within the chapter
3. ✅ Verify the full screen player next chapter button progress is working

## Screencast 

https://github.com/Automattic/pocket-casts-android/assets/308331/c768e980-d675-418d-94a5-3a817f6e1ee9

### CODE REVIEW

File: modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/Chapter.kt [hunk 1]
```diff
@@ -5,14 +5,12 @@ import kotlin.math.roundToInt
...
 data class Chapter(
     val title: String,
-    val url: HttpUrl?,
     var startTime: Int,
     var endTime: Int,
-    val imagePath: String?,
-    val mimeType: String?,
-    var played: Boolean = false,
-    var index: Int = 0,
-    var progress: Float = 0.0f
```
Comment (geekygecko) [inline modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/to/Chapter.kt:15]: It didn't feel right storing the `progress` and `played` state in the chapter object as they can change regularly based on the playback position.

**Comment by CookieyedCodes** [comment]
Chapters might be currently on the current rc, will this also fix it? Someone on Reddit pointed it out to me 🤔

Also as the chapter code is being looked at any further thoughts for RSS timetags to "chapter conversation" for non chaptered podcasts?

**Comment by geekygecko** [comment]
@CookieyedCodes 

> Chapters might be currently on the current rc, will this also fix it? Someone on Reddit pointed it out to me 🤔

This bug is fixed in 7.50 and the open beta version will be out early this week.

> Also as the chapter code is being looked at any further thoughts for RSS timetags to "chapter conversation" for non chaptered podcasts?

I did notice a podcast on the weekend which had timestamps in the show notes but no chapters. It would be a handy feature. When we are adding Podlove and Podcast Index chapters we should do this. Thanks for the reminder, again. One day it will happen. 😄

**Comment by CookieyedCodes** [comment]
I'll keep an eye out ❤️❤️

**Review by ashiagr — APPROVED** [review]
Works well in my testing 👍

**Pull request merged**